comms/EclMeta.h

```diff
@@ -49,33 +49,91 @@ class CEclFolder : public CEclMetaData
     }
 };
 
+class CEclParam : public CEclMetaData
+{
+protected:
+    std::_tstring m_name;
+    std::_tstring m_type;
+public:
+    BEGIN_CUNKNOWN
+    END_CUNKNOWN(CUnknown)
+
+    CEclParam() { }
+    std::_tstring GetName() { return m_name; }
+    std::_tstring GetType() { return m_type; }
+    void UpdateAttrs(Element &e)
+    {
+        GetValue(e, _T("name"), m_name);
+        GetValue(e, _T("type"), m_type);
+    };
+    void AddType(Element e)
+    {
+        GetValue(e, _T("type"), m_type);
+    }
+    void AddName(Element e)
+    {
+        GetValue(e, _T("name"), m_name);
+    }
+};
+typedef StlLinked<CEclParam> CEclParamAdapt;
+typedef std::map<std::wstring, StlLinked<CEclParam> > EclParamMap;
+
 class CEclDefinition : public CEclMetaData
 {
 protected:
     int m_body, m_end, m_line, m_start;
+    std::_tstring m_fullname;
     std::_tstring m_name;
     std::_tstring m_type;
+    std::_tstring m_inheritType;
+    std::_tstring m_functionType;
+    std::_tstring m_content;
     std::map<std::wstring, StlLinked<CEclDefinition> > m_defs;
+    EclParamMap m_params;
+
 public:
     BEGIN_CUNKNOWN
-
     END_CUNKNOWN(CUnknown)
+
     std::map<std::wstring, std::wstring> m_fields;
 
     CEclDefinition() { m_body = m_end = m_line = m_start = 0; }
-    CEclDefinition(const CEclDefinition & def) { m_name = def.m_name; m_type = def.m_type; m_body = def.m_body; m_end = def.m_end;  m_line = def.m_line; m_start = def.m_start; m_fields = def.m_fields; m_defs = def.m_defs; }
+    CEclDefinition(const CEclDefinition & def) { m_fullname = def.m_fullname; m_name = def.m_name; m_type = def.m_type; m_inheritType = def.m_inheritType; m_body = def.m_body; m_end = def.m_end; m_line = def.m_line; m_start = def.m_start; m_fields = def.m_fields; m_defs = def.m_defs; m_params = def.m_params; m_content = def.m_content; m_functionType = def.m_functionType; }
     std::_tstring GetName() { return m_name; }
+    std::_tstring GetFullname() { return m_fullname; }
     std::_tstring GetType() { return m_type; }
+    std::_tstring GetFunctionType() { return m_functionType; }
+    std::_tstring GetInheritType() { return m_inheritType; }
+    std::_tstring GetContent() { return m_content; }
+    EclParamMap GetParams() { return m_params; }
     void UpdateAttrs(Element &e)
     {
         std::wstring name, type;
+        GetValue(e, _T("fullname"), m_fullname);
         GetValue(e, _T("name"), m_name);
         GetValue(e, _T("type"), m_type);
+        GetValue(e, _T("inherittype"), m_inheritType);
         GetIntValue(e, _T("body"), m_body);
         GetIntValue(e, _T("end"), m_end);
         GetIntValue(e, _T("line"), m_line);
         GetIntValue(e, _T("start"), m_start);
     }
+    bool HasDefinitions()
+    {
+        return m_defs.size() ? true : false;
+    }
+    void AddFunctionType(Element e)
+    {
+        GetValue(e, _T("type"), m_functionType);
+    }
+    void AddDocumentation(Element e)
```

Comment:
`const Element & e`?

Reply:
@dehilsterlexis bump

Reply:
@dehilsterlexis bump II

Reply:
@Gordon so sorry! I somehow lost that.  My bad.

Reply:
@GordonSmith not being able to get one const Element &e to work. It is in the AddDocumentation. Not able to solve this one.
![image](https://user-images.githubusercontent.com/10133951/59520649-6f994680-8e98-11e9-9053-fcde217eee0f.png)